rule_packages/cpp/Representation.json

```diff
@@ -53,6 +53,18 @@
           "tags": [
             "correctness"
           ]
+        },
+        {
+          "description": "Passing an aliased pointer to a conceptually restrict-qualified parameter is undefined behavior.",
```

Comment:
Could you please clarify what you mean by "conceptually"?

Reply:
was mostly referring to the fact that the `restrict` keyword comes from C (not C++) so technically this rule means something slightly different in C++ but checks the same concept

ie - in the C version of the [memcpy description](https://en.cppreference.com/w/c/string/byte/memcpy) - overlapping arrays are the problem (but using those violates the restrict contract)

in the C++ [memcpy description](https://en.cppreference.com/w/cpp/string/byte/memcpy) there is no restrict contract broken, but it is just that the arrays overlap

I think I can just reword this to make it clearer for the C++ version that its really only about overlapping arrays